### PR TITLE
Read the fast gas and link native from the highest check block

### PR DESCRIPTION
**Pull request by ferglor**
In this PR, we're adding the highest check block read of native link and fast gas from the keepers 2.1 alpha branch

The idea here is to reduce the number of changes in the alpha branch, and so this discrete piece of functionality is being merged into develop on its own, reducing the impact of the keepers 2.1 alpha branch

### CODE REVIEW

File: core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go [hunk 1]
```diff
@@ -58,15 +58,22 @@ func (enc EVMAutomationEncoder21) Encode(results ...ocr2keepers.CheckResult) ([]
...
 	for i, result := range results {
 		ext, ok := result.Extension.(EVMAutomationResultExtension21)
 		if !ok {
 			return nil, fmt.Errorf("unexpected check result extension struct")
 		}
 
-		// only take these values from the first result
-		// TODO: find a new way to get these values
-		if i == 0 {
+		checkBlock, ok := big.NewInt(0).SetString(string(result.Payload.CheckBlock), 10)
```
Comment (EasterTheBunny) [inline core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go:70]: We are eliminating `CheckBlock` from the payload in favor of the check block number in the `Trigger`

**Review by EasterTheBunny — CHANGES_REQUESTED** [review]
I think just changing checkblock to use the check block number in the trigger is all that needs to be updated

File: core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go (hunk 1 above)
Comment (ferglor) [reply core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go:70]: Oh! So `result.Payload.Trigger.BlockNumber`?

**Review by EasterTheBunny — APPROVED** [review]

File: core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go [hunk 2]
```diff
@@ -58,15 +58,19 @@ func (enc EVMAutomationEncoder21) Encode(results ...ocr2keepers.CheckResult) ([]
 		PerformDatas: make([][]byte, len(results)),
 	}
 
+	fmt.Printf("[EVMAutomationEncoder21] encoding %d results\n", len(results))
```
Comment (amirylm) [inline core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go:61]: remove debug log

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**